### PR TITLE
Updates uthash and makes some tweaks

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: src/uthash.h [hunk 1]
```diff
@@ -714,37 +699,28 @@ typedef unsigned char uint8_t;
         hashv += (unsigned)(keylen);                                                                         \
         switch (_hj_k) {                                                                                     \
             case 11:                                                                                         \
-                hashv += ((unsigned)_hj_key[10] << 24);                                                      \
-                H5_ATTR_FALLTHROUGH                                                                          \
+                hashv += ((unsigned)_hj_key[10] << 24); /* FALLTHROUGH */                                    \
```
Comment (qkoziol) [inline src/uthash.h:702]: Did you want to re-insert the H5_ATTR_FALLTHROUGH?

**Review by qkoziol — APPROVED** [review]

File: src/uthash.h (hunk 1 above)
Comment (derobins) [reply src/uthash.h:702]: No, he uses /* FALLTHROUGH */ comments, which most compilers accept as intent. This is mentioned in the changelogs.

File: src/uthash.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright (c) 2003-2018, Troy D. Hanson     http://troydhanson.github.com/uthash/
+Copyright (c) 2003-2022, Troy D. Hanson  https://troydhanson.github.io/uthash/
```
Comment (derobins) [inline src/uthash.h:2]: The changes in this file are due to the new version of uthash.h

File: src/uthash.h (hunk 1 above)
Comment (qkoziol) [reply src/uthash.h:702]: OK, great

**Review by jhendersonHDF — APPROVED** [review]

**Comment by lrknox** [comment]
CI checks passed in action https://github.com/HDFGroup/hdf5/actions/runs/2731983414 prior to reformatting.

**Review by fortnern — APPROVED** [review]

**Pull request merged**